addons/loadout/functions/fnc_getWeaponList.sqf

```diff
@@ -0,0 +1,42 @@
+#include "script_component.hpp"
+/*
+ * Author: NeilZar
+ * Get all turret weapons from the passed vehicle, the compatible magazines, and a count of each magazine currently in the turret.
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ *
+ * Return Value:
+ * Array of all weapons and their magazines <ARRAY>
+ *
+ * Example:
+ * [_vehicle] call zen_loadout_fnc_getWeaponList
+ *
+ * Public: No
+ */
+
+params ["_vehicle"];
+
+private _pylons = (getPylonMagazines _vehicle) apply {toLower (getText (configFile >> "CfgMagazines" >> _x >> "pylonWeapon"))};
+private _cfgWeapons = configFile >> "CfgWeapons";
+private _allMagazines = magazinesAllTurrets _vehicle;
+
+private _result = [];
+{
+    private _turret = _x;
+    {
+        if (getText (_cfgWeapons >> _x >> "displayName") != "" && (([_x, true] call BIS_fnc_returnParents) arrayIntersect BLACKLIST_WEAPONS isEqualTo [])) then {
```

Comment:
Consider storing the weapon config at the start of the loop also since it is used below, not sure what is better. Also, maybe multi-line this condition.